src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java

```diff
@@ -137,14 +141,32 @@ public SampleFiles getSampleFiles(Long sampleId, Long projectId) {
 			project = projectService.read(projectId);
 		}
 
-		List<SequencingObject> filePairs = getPairedSequenceFilesForSample(sample, project);
-		List<SequencingObject> singles = getSingleEndSequenceFilesForSample(sample, project);
+		List<PairedEndSequenceFileModel> filePairs = getPairedSequenceFilesForSample(sample, project).stream()
+				.map(pair -> new PairedEndSequenceFileModel((SequenceFilePair) pair))
+				.collect(Collectors.toList());
+		List<SingleEndSequenceFileModel> singles = getSingleEndSequenceFilesForSample(sample, project).stream()
+				.map(single -> new SingleEndSequenceFileModel((SingleEndSequenceFile) single))
+				.collect(Collectors.toList());
 		List<SequencingObject> fast5 = getFast5FilesForSample(sample);
 		List<GenomeAssembly> genomeAssemblies = getGenomeAssembliesForSample(sample);
 
 		return new SampleFiles(singles, filePairs, fast5, genomeAssemblies);
 	}
 
+	/**
+	 * Get details about the files belonging to a list of samples
+	 *
+	 * @param ids       - List of sample identifiers to get file details for
+	 * @param projectId - the project id that these samples belong to
+	 * @return A map of sample id and their related file information
+	 */
+	public SampleFilesResponse getFilesForSamples(List<Long> ids, Long projectId) {
```

Comment:
Can you change `ids` to `sampleIds`?

Reply:
Updated in c9dd63576bd3e55230ed863150f357be7f1f34c3